backend/src/bundles/users/user.entity.ts

```diff
@@ -9,78 +9,93 @@ class UserEntity implements Entity {
 
     private 'passwordSalt': string;
 
+    private 'fullName'?: string | undefined;
+
     private constructor({
         id,
         email,
         passwordHash,
         passwordSalt,
+        fullName,
     }: {
         id: number | null;
         email: string;
         passwordHash: string;
         passwordSalt: string;
+        fullName?: string | undefined;
     }) {
         this.id = id;
         this.email = email;
         this.passwordHash = passwordHash;
         this.passwordSalt = passwordSalt;
+        this.fullName = fullName;
     }
 
     public static initialize({
         id,
         email,
         passwordHash,
         passwordSalt,
+        fullName,
     }: {
         id: number;
         email: string;
         passwordHash: string;
         passwordSalt: string;
+        fullName?: string | undefined;
```

Comment:
```suggestion
        fullName: string | null;
```

Reply:
I've tried this but then I've encountered the mistyping error between UserModel and UserEntity and `fullName?: string | undefined;` was a workaround for me.
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/93663923/549c8a84-1de5-4c51-917f-65cf2aab6a89)
Here the returned type is `Promise<UserEntity[]>`.

Or everything needs to be updated so that user was always fetched and returned with user details alongside. But it is not always needed

Reply:
```js
intialize({
    ...user
    fullName: user.fullName ?? null
})
```

Reply:
and `it` better to rename to `user`